src/nakamoto-signer-developer-guide.md

```diff
@@ -0,0 +1,196 @@
+
+## Prerequisites
+
+Rust - [To install](https://www.rust-lang.org/tools/install).
+
+Accessible Stacks node - A Stacks node running the stackerDB instance which is used for signer communication and transaction monitoring and broadcasting.
+
+Accessible Bitcoin node - A Bitcoin node used for transaction monitoring and broadcasting.
+
+A Stacks Address - Identify your address as a Signer
+
+## Installing
+
+### Building from Source
+
+If you wish to compile the default binary from source, follow the steps outlined below. Otherwise, download the binary directly (below)
+
+1. First, clone the Stacks sBTC mono repository:
+
+```console
+git clone git@github.com:stacks-network/stacks-blockchain.git
+```
+
+2. Next, navigate to the stacks-signer directory:
+
+```console
+cd stacks-blockchain/stacks-signer
+```
+
+3. Checkout the appropriate release branch you wish to use if you are not using the default main branch
+
+```console
+git checkout main
+```
+
+4. Compile the signer binary:  
+   Note the binary path defaults to `target/release/stacks-signer`.
+
+```console
+cargo build --release
+```
+
+### Downloading the Binary
+
+1. First, download the precompiled default [TODO:NEED:LINK](LINK).
+
+2. Untar the file
+
+```console
+tar -xvf signer_binary.tar
+```
+
+3.  Check Extracted Files:
+    After running the untar command, the contents of the tar file should be extracted to the current directory. You should see the signer binary (stacks-signer-mini) and the configuration file (signer.toml) listed among the extracted files.
+
+4.  Next, install the signer.
+
+```console
+cargo install --path stacks-signer
+```
+
+## Configuration
+
+The signer takes a TOML config file with the following expected properties
+
+| Key                         | Required | Description                                                                                                                                                  |
+| --------------------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `signer_private_key`        | `true`   | Stacks private key of the signer, used for signing sBTC transactions.                                                                                        |
+| `stacks_node_rpc_url`       | `true`   | Stacks node RPC URL that points to a node running the stackerDB instance which is used for signer communication and transaction monitoring and broadcasting. |
+| `bitcoin_node_rpc_url`      | `true`   | Bitcoin node RPC URL used for transaction monitoring and broadcasting.                                                                                       |
+| `revealer_rpc_url`          | `true`   | Revealer RRC URL                                                                                                                                             |
```

Comment:
This revealer_rpc_url is going away. Can remove it.

Reply:
fixed 97695ec9ce37137f9836f3e62bd22c7701fb51de